### PR TITLE
mysql, types: move `MyDecimal` and `Time` from 'mysql' to 'types'

### PR DESCRIPTION
**Pull request by coocood**
So we don't need to handle sql mode in 'mysql' package, avoid potential dependency cycle.

### CODE REVIEW

**Comment by coocood** [comment]
@zimulala @hanfei1991 PTAL

**Comment by shenli** [comment]
LGTM

File: util/types/helper.go [hunk 1]
```diff
@@ -126,3 +127,68 @@ func CalculateSum(sum Datum, v Datum) (Datum, error) {
...
+
+func isSpace(c byte) bool {
+	return c == ' ' || c == '\t'
+}
+
+func isDigit(c byte) bool {
+	return c >= '0' && c <= '9'
+}
+
+func myMax(a, b int) int {
```
Comment (coocood) [inline util/types/helper.go]: Used by MyDecimal

**Comment by hanfei1991** [comment]
LGTM

**Pull request merged**